gnomad/utils/constraint.py

```diff
@@ -360,3 +412,279 @@ def collapse_strand(
+    :param pops: List of populations used to build coverage and plateau models.
+      Default is ().
+    :param keys: Keys to group observed and possible variant counts.
+      Default is ("context","ref", "alt", "methylation_level", "mu_snp").
+    :param cov_cutoff: Median coverage cutoff. Sites with coverage above this cutoff
+      are considered well covered and will be used to build plateau models. Sites below
+      this cutoff have low coverage and will be used to build coverage models. Defaults
+      to `COVERAGE_CUTOFF`.
+    :param coverage_model: Whether to build a coverage model. Default is False.
+    :return: Coverage model and plateau models.
+    """
+    # Filter to sites with coverage above `cov_cutoff`.
+    high_cov_ht = coverage_ht.filter(coverage_ht.exome_coverage >= cov_cutoff)
+    agg_expr = {
+        "observed_variants": hl.agg.sum(high_cov_ht.observed_variants),
+        "possible_variants": hl.agg.sum(high_cov_ht.possible_variants),
+    }
+    for pop in pops:
+        agg_expr[f"observed_{pop}"] = hl.agg.array_sum(
+            high_cov_ht[f"downsampling_counts_{pop}"]
+        )
+
+    # Generate a Table with all necessary annotations (x and y listed above)
+    # for the plateau models.
+    high_cov_group_ht = high_cov_ht.group_by(*keys).aggregate(**agg_expr)
+    high_cov_group_ht = annotate_mutation_type(high_cov_group_ht)
+
+    # Build plateau models.
+    plateau_models_agg_expr = build_plateau_models(
+        cpg_expr=high_cov_group_ht.cpg,
+        mu_snp_expr=high_cov_group_ht.mu_snp,
+        observed_variants_expr=high_cov_group_ht.observed_variants,
+        possible_variants_expr=high_cov_group_ht.possible_variants,
+        pop_observed_variants_exprs={
+            pop: high_cov_group_ht[f"observed_{pop}"] for pop in pops
+        },
+        weighted=weighted,
+    )
+    plateau_models = high_cov_group_ht.aggregate(hl.struct(**plateau_models_agg_expr))
+
+    coverage_model = None
+    if build_coverage_model:
+        # Filter to sites with coverage below `cov_cutoff` and larger than 0.
+        low_cov_ht = coverage_ht.filter(
+            (coverage_ht.exome_coverage < cov_cutoff) & (coverage_ht.exome_coverage > 0)
+        )
+
+        # Metric that represents the relative mutability of the exome calculated on high
+        # coverage sites
+        # This metric is used as scaling factor when building the coverage model.
+        high_coverage_scale_factor = high_cov_ht.aggregate(
+            hl.agg.sum(high_cov_ht.observed_variants)
+            / hl.agg.sum(high_cov_ht.possible_variants * high_cov_ht.mu_snp)
+        )
+
+        # Generate a Table with all necessary annotations (x and y listed above)
+        # for the coverage model.
+        low_cov_group_ht = low_cov_ht.group_by(
+            log_coverage=hl.log10(low_cov_ht.exome_coverage)
+        ).aggregate(
+            low_coverage_obs_exp=hl.agg.sum(low_cov_ht.observed_variants)
```

Comment:
```suggestion
            low_coverage_oe=hl.agg.sum(low_cov_ht.observed_variants)
```

Reply:
(for consistency, since `build_coverage_model` now uses `oe` rather than `obs_exp`)